app/src/main/java/com/gatheringhallstudios/mhworlddatabase/features/items/ItemListFragment.kt

```diff
@@ -35,6 +35,17 @@ class ItemListDelegate(private val onSelect: (ItemView) -> Unit) : SimpleListDel
 }
 
 class ItemListFragment : Fragment() {
+    companion object {
+        private val ARG_CATEGORY = "CATEGORY"
+
+        @JvmStatic
+        fun newInstance(category: ItemCategory): ItemListFragment {
+            val f = ItemListFragment()
+            f.arguments = BundleBuilder().putSerializable(ARG_CATEGORY, category).build()
```

Comment:
Best practice is to avoid putting Serializable into Bundles. use putString or something more explicit. If it's an object then it needs to implement [Parcelable](https://stackoverflow.com/questions/3323074/android-difference-between-parcelable-and-serializable), which is a pain in the ass to implement by hand, but easier with the [Parceler](https://github.com/johncarl81/parceler) library.

Reply:
We spoke about this a little more. Since its an Enum its fine to use putSerializable.

I'm tempted on making an extension functions to make working with enums easier.